swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/VPNConfigurationManager.swift

```diff
@@ -7,82 +7,41 @@
-    case .decodeIPCDataFailed:
-      return "Decoding IPC data failed."
-    case .invalidNotification:
-      return "NEVPNStatusDidChange notification doesn't seem to be valid."
-    case .cannotLoad:
-      return "Could not load VPN configurations!"
-    case .noIPCData:
-      return "No IPC data returned from the XPC connection!"
-    case .invalidStatus(let status):
-      return "The IPC operation couldn't complete because the VPN status is \(status)."
+public class VPNConfigurationManager {
+  enum Error: Swift.Error {
+    case managerNotInitialized
+    case savedProtocolConfigurationIsInvalid
+
+    var localizedDescription: String {
+      switch self {
+      case .managerNotInitialized:
+        return "NETunnelProviderManager is not yet initialized. Race condition?"
+      case .savedProtocolConfigurationIsInvalid:
+        return "Saved protocol configuration is invalid. Check types?"
+      }
     }
   }
-}
-
-public enum VPNConfigurationManagerKeys {
-  static let actorName = "actorName"
-  static let authBaseURL = "authBaseURL"
-  static let apiURL = "apiURL"
-  public static let accountSlug = "accountSlug"
-  public static let logFilter = "logFilter"
-  public static let internetResourceEnabled = "internetResourceEnabled"
-}
 
-// TODO: Refactor this to remove the lint ignore
-// swiftlint:disable:next type_body_length
-public class VPNConfigurationManager {
-
-  // Connect status updates with our listeners
-  private var tunnelObservingTasks: [Task<Void, Never>] = []
-
-  // Track the "version" of the resource list so we can more efficiently
-  // retrieve it from the Provider
-  private var resourceListHash = Data()
-
-  // Cache resources on this side of the IPC barrier so we can
-  // return them to callers when they haven't changed.
-  private var resourcesListCache: ResourceList = ResourceList.loading
+  public enum Keys {
+    static let actorName = "actorName"
+    static let authBaseURL = "authBaseURL"
+    static let apiURL = "apiURL"
+    public static let accountSlug = "accountSlug"
+    public static let logFilter = "logFilter"
+    public static let internetResourceEnabled = "internetResourceEnabled"
+  }
 
   // Persists our tunnel settings
-  private var manager: NETunnelProviderManager?
-
-  // Indicates if the internet resource is currently enabled
-  public var internetResourceEnabled: Bool = false
-
-  // Encoder used to send messages to the tunnel
-  private let encoder = {
-    let encoder = PropertyListEncoder()
-    encoder.outputFormat = .binary
-
-    return encoder
-  }()
+  var manager: NETunnelProviderManager?
 
   public static let bundleIdentifier: String = "\(Bundle.main.bundleIdentifier!).network-extension"
-  private let bundleDescription = "Firezone"
+  static let bundleDescription = "Firezone"
 
   // Initialize and save a new VPN configuration in system Preferences
-  func create() async throws {
+  static func create() async throws -> NETunnelProviderManager {
```

Comment:
Why doesn't this just return `VPNConfigurationManager`?

Reply:
This is basically a ctor isn't it?

Reply:
Yes, essentially. However, it has side effects that seem unintuitive for an `init()` call to have. Namely, it prompts the user for VPN permission, opening System Settings.app, and then installing the VPN configuration.

I've started to realize it's probably a good idea to keep `init()` as lightweight as possible in Swift. That allows you to instantiate the view / state hierarchy (i.e. launch the app) with as little state as necessary, and then update needed properties on those instances by populating them after launch.